src/main/java/org/aksw/simba/squirrel/components/DeduplicatorComponent.java

```diff
@@ -0,0 +1,224 @@
+                }
+            } else {
+                LOGGER.warn("Couldn't get {} from the environment. An in-memory queue will be used.", FrontierComponent.RDB_HOST_NAME_KEY);
+            }
+
+            if ((rdbHostName != null) && (rdbPort > 0)) {
+                knownUriFilter = new RDBKnownUriFilter(rdbHostName, rdbPort, FrontierComponent.RECRAWLING_ACTIVE);
+                knownUriFilter.open();
+            } else {
+                knownUriFilter = new InMemoryKnownUriFilter(FrontierComponent.RECRAWLING_ACTIVE);
+            }
+
+            // TODO: other kinds of sinks must be possible as well
+            sink = new SparqlBasedSink(SparqlUtils.getDatasetUriForUpdate(), SparqlUtils.getDatasetUriForQuery());
+
+            serializer = new GzipJavaUriSerializer();
+
+            try {
+                RabbitQueue rabbitQueue = this.incomingDataQueueFactory.createDefaultRabbitQueue(DEDUPLICATOR_QUEUE_NAME);
+                receiver = DataReceiverImpl.builder().dataHandler(this)
+                    .maxParallelProcessedMsgs(100).queue(rabbitQueue).build();
+
+                senderFrontier = DataSenderImpl.builder().queue(outgoingDataQueuefactory, FrontierComponent.FRONTIER_QUEUE_NAME)
+                    .build();
+            } catch (IOException e) {
+                LOGGER.error("Error while creating sender object.", e);
+            }
+        }
+    }
+
+    @Override
+    public void run() {
+
+    }
+
+    /**
+     * Compare the hash values of the uris in {@link #newUrisBufferSet} with the hash values of all uris contained
+     * in {@link #knownUriFilter}.
+     */
+    private void compareNewUrisWithOldUris() {
+        List<CrawleableUri> allUris = knownUriFilter.getAllUris();
+        Set<CrawleableUri> set = new HashSet<>(allUris);
+        set.addAll(newUrisBufferSet);
+
+        outer:
+        for (CrawleableUri uriNew : newUrisBufferSet) {
+            for (CrawleableUri uriOld : set) {
+                if (!uriOld.equals(uriNew)) {
+                    if (uriOld.getHashValue().equals(uriNew.getHashValue())) {
+                        // get triples from pair1 and pair2 and compare them
+                        Set<Triple> setOld = new HashSet<>(sink.getTriplesForGraph(uriOld));
+                        Set<Triple> setNew = new HashSet<>(sink.getTriplesForGraph(uriNew));
+
+                        boolean equal = true;
+                        for (Triple triple : setOld) {
+                            if (!setNew.contains(triple)) {
+                                equal = false;
+                                break;
+                            }
+                        }
+
+                        if (equal) {
+                            // TODO: delete duplicate
+                            continue outer;
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Summarizing this method: I think that it could work the way you have implemented it. However, there might be a more elegant way to handle it. Define two interface
1. for calculating hashes for new datasets
1. for retrieving URIs which are candidates for comparison (i.e., that have the same hash)
2. for checking whether two datasets contain the same triples

Disadvantages:
I can not see any real disadvantages. Maybe the retrieval of datasets with the same hash becomes a little bit more difficult since you might not be able to ask Redis for all URIs with a certain hash. On the other hand side you don't want to load all URIs from Redis again and again :thinking: 

Advantages:
Both are decoupled from the main implementation and can be easily exchanged. For example: if we are using the SPARQL sink in the future (especially for storing the meta data of crawled URIs), we can directly ask the SPARQL endpoint for URIs with the same hash (without iterating over all possible results). Additionally, an easy comparison of two graphs is possible via SPARQL.